### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | /
| Type?         | improvement
| Category?     | CO
| BC breaks?    |  no
| Deprecations? | no
| Fixed ticket? | /
| How to test?  | Everything should work the same.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11639)

### CODE REVIEW

**Comment by MathiasReker** [comment]
I need help to solve those conflicts, I can not do it from the web editor :/

File: tools/build/Library/InstallUnpacker/classes/VersionNumber.php [hunk 1]
```diff
@@ -115,7 +115,7 @@ public function __toString()
      *
      * @return int 1 if this version number is higher, -1 if lower, 0 if equal
      */
-    public function compare(VersionNumber $otherNumber)
+    public function compare(self $otherNumber)
```
Comment (mickaelandrieu) [inline tools/build/Library/InstallUnpacker/classes/VersionNumber.php]: not sure it is accepted in all versions of PHP, I've never seen this kind of use before: would you mind to revert it?

**Review by mickaelandrieu — CHANGES_REQUESTED** [review]

File: tools/build/Library/InstallUnpacker/classes/VersionNumber.php (hunk 1 above)
Comment (MathiasReker) [reply tools/build/Library/InstallUnpacker/classes/VersionNumber.php]: Sure

Comment (MathiasReker) [reply tools/build/Library/InstallUnpacker/classes/VersionNumber.php]: Fixed.

**Comment by MathiasReker** [comment]
I made another PR instead: https://github.com/PrestaShop/PrestaShop/pull/11658

**Pull request closed without merging**